sensor/common/collectorruntimeconfig/service_impl.go

```diff
@@ -0,0 +1,163 @@
+package collectorruntimeconfig
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/grpc/authz/idcheck"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/message"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// CollectorService is the struct that manages the collector configuration
+type serviceImpl struct {
+	sensor.UnimplementedCollectorServiceServer
+
+	collectorC chan common.MessageToCollectorWithAddress
```

Comment:
`collectorC <-chan common.MessageToCollectorWithAddress` resulted in a build error. I am not clear why.